packages/api/src/providers/oauth1.ts

```diff
@@ -161,23 +161,26 @@ export class OAuth1DefaultImpl implements OAuth1 {
     return { oauth_token, oauth_token_secret };
   }
 
-  async deregister(userAccessTokens: string[]): Promise<void> {
+  async deregister(userAccessTokens: string[], cxId?: string): Promise<void> {
     for (const oauthUserAccessToken of userAccessTokens) {
       const userTokenList = await getUserTokenByUAT({ oauthUserAccessToken });
       for (const userToken of userTokenList) {
-        capture.setUser({ id: userToken.userId });
         // DynamoDB (Webhook and auth)
-        const updatedUserToken = userToken.clone();
-        updatedUserToken.oauthUserAccessToken = undefined;
-        updatedUserToken.oauthUserAccessSecret = undefined;
-        await saveUserToken(updatedUserToken);
-        // Postgres (app standard)
-        await updateProviderData({
-          id: userToken.userId,
-          cxId: userToken.cxId,
-          provider: this.providerName,
-          providerItem: undefined,
-        });
+        if (!cxId || cxId === userToken.cxId) {
+          capture.setUser({ id: userToken.userId });
```

Comment:
I know this was already in place, but this caught my attention: we use Sentry's `user` to represent our cx, not our cx's users. I think it was like this b/c we didn't have the `cxId` for some of these OAuth flows, but I see we do have it for OAuth1.

Maybe we just set it based on `userToken.cxId`? Could keep storing the `userId` by exposing a `setExtra` on `capture`, which calls `Sentry.setExtra()`, and calling `capture.setExtra({ userId: userToken.userId });`.

Could check if that's the case for OAuth2 as well.

Reply:
Our capture.setUser only takes an `id` and/or and `email` as argument's attributes.
And .setExtra() does not exist on `capture`.  

I could write something like:
`capture.setUser({ id: userToken.cxId });` 
but not sure if that's appropriate

Reply:
Look at this part of my comment:

> ...exposing a `setExtra` on `capture`, which calls `Sentry.setExtra()`...

"exposing": create a public function

In addition to what you suggested, so we set the user ID as the `cxId`.

Reply:
Gotcha!